packages/sentinel_one/data_stream/activity/elasticsearch/ingest_pipeline/default.yml

```
@@ -23,12 +14,74 @@ processors:
       target_field: _id
       ignore_missing: true
-  - date: 
+  - script:
+      description: Set the value of event.category, event.outcome, event.type and event.kind.
+      lang: painless
+      source: >
+          def eventCategory = new HashSet();
+          def eventType = new HashSet();
+          if (ctx.json?.threatId != null && ctx.json?.threatId != '') {
+              def threat_classification = ctx.json?.data?.threatClassification;
+              if (threat_classification == 'Exploit' || threat_classification == 'PUA') {
+                  eventCategory.add('threat');
+                  ctx.event.type = ['indicator'];
+              } else if (threat_classification == 'Malware' || threat_classification == 'Ransomware' || threat_classification == 'Trojan' || threat_classification == 'Downloader') {
+                  eventCategory.add('malware');
+                  ctx.event.type = ['info'];
+              }
+          }
+          else if (ctx.json?.primaryDescription != null) {
+              def description = ctx.json.primaryDescription.toLowerCase();
+              if (description.contains('logged in')) {
+                  eventCategory.add('authentication');
+                  eventType.add('start');
+                  ctx.event.outcome = 'success';
+              } else if (description.contains('logged out')) {
+                  eventCategory.add('authentication');
+                  eventType.add('end');
+              }
+              if (description.contains('created') || description.contains('added')) {
+                  eventCategory.add('configuration');
+                  eventType.add('creation');
+              }
+              if (description.contains('deleted')) {
+                  eventCategory.add('configuration');
+                  eventType.add('deletion');
+              }
+              if (description.contains('edited') || description.contains('updated') || description.contains('modified')) {
+                  eventCategory.add('configuration');
+                  eventType.add('change');
+              }
+              if (description.contains('enabled') || description.contains('recovery email')) {
+                  eventCategory.add('configuration');
+              }
+              if (description.contains('verification email')) {
+                  eventCategory.add('email');
+              }
+              if (description.contains('failed to log in')) {
+                  eventCategory.add('authentication');
+                  ctx.event.outcome = 'failure';
+              }
+              if (!eventType.isEmpty()) {
+                  ctx.event.type = eventType;
+              } else {
+                  ctx.event.type = ['info'];
+              }
+          }
+          if (!eventCategory.isEmpty()) {
+              ctx.event.category = eventCategory;
+          }
+          if (ctx.json?.data?.confidenceLevel != null && ['suspicious', 'malicious'].contains(ctx.json?.data?.confidenceLevel)) {
```

`[...].contains(null)` returns false unless the array list holds a `null`, so this can be rewritten:

```suggestion
          if (['suspicious', 'malicious'].contains(ctx.json?.data?.confidenceLevel)) {
```